lib/discordrb.rb

```diff
@@ -48,9 +48,9 @@ def self.split_message(msg)
     # Find the largest element that is still below the character limit, or if none such element exists return the first
     ideal = joined.max_by { |e| e.length > CHARACTER_LIMIT ? -1 : e.length }
 
-    # If it's still larger than the character limit (none was smaller than it) split it into slices with the length
-    # being the character limit, otherwise just return an array with one element
-    ideal_ary = ideal.length > CHARACTER_LIMIT ? ideal.chars.each_slice(CHARACTER_LIMIT).map(&:join) : [ideal]
+    # If it's still larger than the character limit (none was smaller than it) split it into the largest chunk without
+    # cutting words apart, breaking on the nearest space within character limit, otherwise just return an array with one element
+    ideal_ary = ideal.length > 2000 ? ideal.split(/(.{,2000}\b+?)/).reject(&:empty?) : [ideal]
```

Comment:
Why is `CHARACTER_LIMIT` hardcoded to 2000 here?!
```suggestion
    ideal_ary = ideal.length > CHARACTER_LIMIT ? ideal.split(/(.{,CHARACTER_LIMIT}\b+?)/).reject(&:empty?) : [ideal]
```

Reply:
Honestly? The way i was doing my algorithmic testing i didn't feel like also having a while line for declaring character limit for, the literal one line of code in the test, so i changed it to 2000.
and, evidently, forgot to change it back before pushing to my repo. That's my bad.